File: GdPicture/WinForm/vb.net/Forms Processing - OMR/OMR/Form1.vb
Option Strict On
Option Explicit On

Imports GdPicture14


Public Class Form1

    Private m_ImageID As Integer
    Private m_GdPictureImaging As New GdPictureImaging



    Private Sub Form1_Load(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles MyBase.Load
        Dim oLicenseManager As New GdPicture14.LicenseManager() 'Go to http://www.gdpicture.com/download-gdpicture/ to get a 1 month trial key unlocking all features of the toolkit.
        oLicenseManager.RegisterKEY("XXXX") 'Please, replace XXXX by a valid demo or commercial license key.
    End Sub



    Private Sub releasegdpictureimage()
        If m_ImageID <> 0 Then
            m_GdPictureImaging.ReleaseGdPictureImage(m_ImageID)
            m_ImageID = 0
        End If
        GdViewer1.CloseDocument()
    End Sub

    Private Sub CloseToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles CloseToolStripMenuItem.Click
        releasegdpictureimage()
    End Sub

    Private Sub ExitToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ExitToolStripMenuItem.Click
        releasegdpictureimage()
        Me.Close()
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        GdViewer1.ZoomIN()
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        GdViewer1.ZoomOUT()
    End Sub

    Private Sub OpenToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OpenToolStripMenuItem.Click
        releasegdpictureimage()
        m_ImageID = m_GdPictureImaging.CreateGdPictureImageFromFile("")
        If m_ImageID <> 0 Then
            GdViewer1.DisplayFromGdPictureImage(m_ImageID)
            GdViewer1.MouseMode = ViewerMouseMode.MouseModeAreaSelection
        End If
    End Sub

    Private Sub GdViewer1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles GdViewer1.Load

    End Sub

    Private Sub GdViewer1_MouseUp(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles GdViewer1.MouseUp
        If GdViewer1.IsRect Then
            Dim Left, Top, Width, Height As Integer
            GdViewer1.GetRectCoordinatesOnDocument(Left, Top, Width, height)
            Dim Rects(1) As Rectangle
            Dim Result() As Integer

            Rects(0).X = Left
            Rects(0).Y = Top
            Rects(0).Width = Width
            Rects(0).Height = Height
            If DefaultToolStripMenuItem.Checked Then
                Result = m_GdPictureImaging.OMRDetectMarks(m_ImageID, Rects, 1)
            Else
                Result = m_GdPictureImaging.OMRDetectOvalMarks(m_ImageID, Rects, 1)
            End If

            MsgBox("Value: " & Str(Result(0)))
        End If
    End Sub

    Private Sub DefaultToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles DefaultToolStripMenuItem.Click
        DefaultToolStripMenuItem.Checked = Not DefaultToolStripMenuItem.Checked
        OvalToolStripMenuItem.Checked = Not DefaultToolStripMenuItem.Checked
    End Sub

    Private Sub OvalToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OvalToolStripMenuItem.Click
        OvalToolStripMenuItem.Checked = Not OvalToolStripMenuItem.Checked
        DefaultToolStripMenuItem.Checked = Not OvalToolStripMenuItem.Checked
    End Sub
End Class

File: GdPicture/WinForm/C#/Forms Processing - OMR/OMR/Form1.cs
using GdPicture14; 
using System;
using System.Collections;
using System.Diagnostics;
using System.Drawing;
using System.Windows.Forms;
namespace adjust
{
    public partial class Form1  
    { 
        public Form1() 
        { 
            InitializeComponent();if (transDefaultFormForm1 == null)	transDefaultFormForm1 = this;
           
        } 
        
        private int m_ImageID; 
        private GdPictureImaging m_GdPictureImaging = new GdPictureImaging(); 
        
        
        
        private void Form1_Load( System.Object eventSender, System.EventArgs eventArgs ) 
        {
            LicenseManager oLicenseManager = new LicenseManager(); //Go to http://www.gdpicture.com/download-gdpicture/ to get a 1 month trial key unlocking all features of the toolkit.
            oLicenseManager.RegisterKEY("XXXX"); //Please, replace XXXX by a valid demo or commercial license key.
        } 
        
 
        private void ReleaseGdpictureImage() 
        { 
            if ( m_ImageID != 0 ) 
            { 
                m_GdPictureImaging.ReleaseGdPictureImage( m_ImageID ); 
                m_ImageID = 0; 
            } 
            GdViewer1.CloseDocument(); 
        } 
        
        
        private void CloseToolStripMenuItem_Click( System.Object sender, System.EventArgs e ) 
        {
            ReleaseGdpictureImage(); 
        } 
        
        
        private void ExitToolStripMenuItem_Click( System.Object sender, System.EventArgs e ) 
        {
            ReleaseGdpictureImage(); 
            this.Close(); 
        } 
        
        
        private void Button1_Click( System.Object sender, System.EventArgs e ) 
        { 
            if(GdViewer1.ZoomIN() != GdPictureStatus.OK)
                MessageBox.Show("Error: " + GdViewer1.GetStat().ToString()); 
        } 
        
        
        private void Button2_Click( System.Object sender, System.EventArgs e ) 
        {
            if (GdViewer1.ZoomOUT() != GdPictureStatus.OK)
                MessageBox.Show("Error: " + GdViewer1.GetStat().ToString()); 
        } 
        

        private void OpenToolStripMenuItem_Click( System.Object sender, System.EventArgs e ) 
        {
            ReleaseGdpictureImage(); 
            m_ImageID = m_GdPictureImaging.CreateGdPictureImageFromFile( "" );
            if (m_ImageID != 0)
            {
                if (GdViewer1.DisplayFromGdPictureImage(m_ImageID) != GdPictureStatus.OK)
                    MessageBox.Show("Cannot Display Image");
                GdViewer1.MouseMode = ViewerMouseMode.MouseModeAreaSelection;
            }
            else
            {
                MessageBox.Show("No Image was loaded");
            }
        } 
        
        
        
        private void GdViewer1_MouseUp( object sender, System.Windows.Forms.MouseEventArgs e ) 
        { 
            if ( GdViewer1.IsRect() ) 
            { 
                int Left = 0, Top = 0, Width = 0, Height = 0; 
                GdViewer1.GetRectCoordinatesOnDocument( ref Left, ref Top, ref Width, ref Height ); 
                Rectangle[] Rects = new Rectangle[ 2 ]; 
                int[] Result = null; 
                
                Rects[ 0 ].X = Left; 
                Rects[ 0 ].Y = Top; 
                Rects[ 0 ].Width = Width; 
                Rects[ 0 ].Height = Height;
                if(defaultToolStripMenuItem.Checked)
                    Result = m_GdPictureImaging.OMRDetectMarks( m_ImageID, Rects, 1 ); 
                else
                    Result = m_GdPictureImaging.OMRDetectOvalMarks(m_ImageID, Rects, 1);
                MessageBox.Show("Value: " + Result[0].ToString());

            } 
        } 
        
        private static Form1 transDefaultFormForm1 = null;
        public static Form1 TransDefaultFormForm1
        { 
        	get
        	{ 
        		if (transDefaultFormForm1 == null)
        		{
        			transDefaultFormForm1 = new Form1();
        		}
        		return transDefaultFormForm1;
        	} 
        }

        private void defaultToolStripMenuItem_Click(object sender, EventArgs e)
        {
            defaultToolStripMenuItem.Checked = !defaultToolStripMenuItem.Checked;
            ovalToolStripMenuItem.Checked = !ovalToolStripMenuItem.Checked;
        }

        private void ovalToolStripMenuItem_Click(object sender, EventArgs e)
        {
            defaultToolStripMenuItem.Checked = !defaultToolStripMenuItem.Checked;
            ovalToolStripMenuItem.Checked = !ovalToolStripMenuItem.Checked;
        } 
    } 
    
} 
